bson/bsonoptions/uint_codec_options.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsonoptions
+
+// UIntCodecOptions represents all possible options for uint encoding and decoding.
+type UIntCodecOptions struct {
+	EncodeToMinSize *bool // Specifies if all uints except uint64 should be decoded to minimum size bsontype. Defaults to false.
```

Comment:
Why not uint64? I mean, just because it's a uint64 in code doesn't mean they want the extra bytes in the database if they aren't necessary.

Reply:
Our actual minsize drops uint64s to int32s too, but this is to match mgo, which by default tries to fit everything but uint64s into an int32. I just didn't have a good name for that specific behavior